### ISSUE TITLE
Function visibility marker Check for PHP and JS Files

### ISSUE THREAD

**Issue by snightingale** [squizlabs/PHP_CodeSniffer#1676]
Hi,

I may have not seen it but I was wondering if there can be a Function visibility marker check for both PHP and JS files?

THis is an example here:

http://php.net/manual/en/language.oop5.visibility.php

Basically to flag up if a function is missing Public, Private or Protected.

THis would help alot and would be a good addition.

**Comment by jrfnl** [comment]
Well, the existing sniffs do not check JS, only PHP, but I get the feeling you are looking for `Squiz.Scope.MethodScope`

For more of the same:
* `Squiz.Scope.MemberVarScope` - same but for properties
* `Squiz.WhiteSpace.ScopeKeywordSpacing` - spacing after scope keyword
* `PSR2.Methods.MethodDeclaration` - modifier keyword order, i.e. `final abstract public static`

**Comment by snightingale** [comment]
Thanks for the reply @jrfnl!

Ah I assumed so but wanted to check to see if that was the case or not. Cool, Squiz.Scope.MethodScope is what im looking for but would need to trigger it for JS.  I have made some Custom Sniffs from other Commenting sniffs already in phpcs but I added the following:

public $supportedTokenizers = array('JS');

Does this also take effect if I made a copy of Squiz.Scope.MethodScope and add it?

I shall take a look at toher properties too, see if that will trigger too.
